soroban-env-host/src/budget.rs

```diff
@@ -118,23 +123,38 @@ impl CostType {
 
 #[derive(Default, Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
 pub struct CostModel {
-    pub const_param: u64,
-    pub lin_param: u64,
+    pub(crate) const_param: u64,
+    pub(crate) lin_param: u64,
+    pub(crate) iterations: u64,
```

Comment:
I don't think `iterations` should be stored in the cost model, should it? I thought it was passed in from the budget-charging call site.

Reply:
Yes, it is now part of the input `f(x) = N * (a + b * Option<x>)` as `N`. 
It is part of the model because `N` is tied to a particular `x`, i.e. charges with different inputs cannot be charged together.